cdap-watchdog-api/src/main/java/co/cask/cdap/api/metrics/MetricDeleteQuery.java

```diff
@@ -18,33 +18,31 @@
 
 import com.google.common.base.Joiner;
 import com.google.common.base.Objects;
-import com.google.common.collect.ImmutableList;
-import com.google.common.collect.Maps;
 
 import java.util.Collection;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.List;
 import java.util.Map;
+import java.util.function.Predicate;
 
 /**
  * Query that specifies parameters to delete entries from {@link MetricStore}.
  */
 public class MetricDeleteQuery {
-
   private final long startTs;
   private final long endTs;
   private final Collection<String> metricNames;
   private final Map<String, String> sliceByTagValues;
+  private final Predicate<List<String>> tagPredicate;
 
   public MetricDeleteQuery(long startTs, long endTs, Collection<String> metricNames,
-                           Map<String, String> sliceByTagValues) {
+                           Map<String, String> sliceByTagValues, List<String> aggregationTags) {
```

Comment:
Please add a javadoc documenting the arguments.

Reply:
Added